core/block.js

```diff
@@ -859,6 +922,33 @@ Blockly.Block.prototype.setColour = function(colour) {
   }
 };
 
+/**
+ * Set the style and colour values of a block.
+ * @param {string} blockStyleName Name of the block style
+ * @throws {Error} if the block style does not exist.
+ */
+Blockly.Block.prototype.setStyle = function(blockStyleName) {
+  var dereferenced = Blockly.utils.replaceMessageReferences(blockStyleName);
+  var style = Blockly.getStyle();
+  if (!style) {
+    throw 'Trying to set block style before Blockly.setStyle() has been called';
+  }
+  var blockStyle = style.getBlockStyle(dereferenced);
+  this.styleName_ = dereferenced;
+
+  if (blockStyle) {
+    this.style_ = blockStyle;
+    this.secondaryColour_ = blockStyle.secondaryColour;
+    this.tertiaryColour_ = blockStyle.tertiaryColour;
+    // Set colour will trigger an updateColour() on a block_svg
+    this.setColour(blockStyle.primaryColour);
+  }
+  else {
+    var errorMsg = 'Invalid style name: ' + blockStyleName;
+    throw errorMsg;
```

Comment:
nit this should be:
throw Error('message');

And you don't need the variable, you can assemble in the Error